src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -489,24 +494,72 @@ private BoundStatement RewriteStringForEachStatement(BoundForEachStatement node)
             return result;
         }
 
+        /// <summary>
+        /// Takes the expression for the current value of the iteration variable and either
+        /// (1) assigns it into a local, or
+        /// (2) deconstructs it into multiple locals (if there is a deconstruct step).
+        ///
+        /// Produces `V v = /* expression */` or `(D1 d1, ...) = /* expression */`.
+        /// </summary>
+        private BoundStatement LocalOrDeconstructionDeclaration(
+                                    BoundForEachStatement forEachBound,
+                                    ForEachStatementSyntax forEachSyntax,
+                                    LocalSymbol iterationVar,
+                                    BoundExpression iterationVarValue,
+                                    out ImmutableArray<LocalSymbol> iterationVariables)
+        {
+            BoundStatement iterationVarDecl;
+            BoundForEachDeconstructStep deconstruction = forEachBound.DeconstructionOpt;
+
+            if (deconstruction == null)
+            {
+                // V v = /* expression */
+                iterationVarDecl = MakeLocalDeclaration(forEachSyntax, iterationVar, iterationVarValue);
+                iterationVariables = ImmutableArray.Create(iterationVar);
+            }
+            else
+            {
+                // (D1 d1, ...) = /* expression */
+                var assignment = deconstruction.DeconstructionAssignment;
+
+                AddPlaceholderReplacement(deconstruction.TargetPlaceholder, iterationVarValue);
+                BoundExpression loweredAssignment = VisitExpression(assignment);
+                iterationVarDecl = new BoundExpressionStatement(assignment.Syntax, loweredAssignment);
+                RemovePlaceholderReplacement(deconstruction.TargetPlaceholder);
+
+                iterationVariables = assignment.LeftVariables.SelectAsArray(v => ((BoundLocal)v).LocalSymbol);
+            }
+
+            return iterationVarDecl;
+        }
+
         private static BoundBlock CreateBlockDeclaringIterationVariable(
             LocalSymbol iterationVariable,
             BoundStatement iteratorVariableInitialization,
             BoundStatement rewrittenBody,
             ForEachStatementSyntax forEachSyntax)
         {
+            return CreateBlockDeclaringIterationVariables(ImmutableArray.Create(iterationVariable), iteratorVariableInitialization, rewrittenBody, forEachSyntax);
+        }
```

Comment:
I thought I got an error when I had removed it, but apparently I was mistaken. It is not used anymore. Removed.